tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+    ''
+    '''
+    url = __get_url(raw_html)
+    res = metahtml.parse(raw_html, url, None, False, extractor_config, True)
+    if res['content'] is not None:
+        cleaned_html = res['content']['best']['value']['text']
+    else:
+        cleaned_html = ''
+    return cleaned_html
+
+def clean_trafilatura_no_fallback(raw_html):
+    '''                                                          
+    Arguments:
+        raw_html        String  Raw downloaded html input
+    Return:
+        cleaned_html    String  Cleaned html by trafilatura library only
+    Faster than with fallback option but less content extracted
+    >>> clean_trafilatura_no_fallback('<div><h3>Test Header</h3><p>Content 123</p></div>')
+    'Content 123'
+    '''
+    url = __get_url(raw_html)
+    cleaned_html = trafilatura.extract(raw_html, url, no_fallback=True)
+    if cleaned_html is None:
+        cleaned_html = ''
+    return cleaned_html
+
+def clean_newspaper3k(raw_html):
+    '''
+    Arguments:
+        raw_html        String  Raw downloaded html input
+    Return:
+        cleaned_html    String  Cleaned html by newspaper3k library
+    '''
+    parser = parsers.Parser()
+    cleaned_html = parser.fromstring(raw_html)
+    return cleaned_html.text_content()
+
+def clean_trafilatura_with_fallback(raw_html):
+    '''
+    Arguments:
+        raw_html        String  Raw downloaded html input
+    Return:
+        cleaned_html    String  Cleaned html by trafilatura library with fallback by justtext and readability-lxml
+    Slower but more content extracted. We opt the doctest for this function as there is a required minimum file size
+    '''
+    url = __get_url(raw_html)
+    cleaned_html = trafilatura.extract(raw_html, url)
+    if cleaned_html is None:
+        cleaned_html = ''
+    return cleaned_html
+
+# Rename function names
+clean_cleaneval.__name__ = 'cleaneval'
+clean_metahtml.__name__ = 'metahtml'
+clean_metahtml_0.__name__ = 'metahtml-0'
+clean_metahtml_1.__name__ = 'metahtml-1'
+clean_metahtml_2.__name__ = 'metahtml-2'
+clean_metahtml_3.__name__ = 'metahtml-3'
+clean_metahtml_4.__name__ = 'metahtml-4'
+clean_metahtml_5.__name__ = 'metahtml-5'
+clean_newspaper3k.__name__ = 'newspaper3k'
+clean_trafilatura_no_fallback.__name__ = 'trafilatura-no-fallback'
+clean_trafilatura_with_fallback.__name__ = 'trafilatura-with-fallback'
+
+
+def cleaneval_from_files(gold_path, raw_path, list_of_libraries = [ lambda x: x ], eval_method=do_cleaneval, opt_latex=True, opt_total=True, opt_noheader=False, opt_summary=True, opt_ascii=False, opt_unlabelled=False):
```

Comment:
rather than `opt_latex` be a boolean variable, have a variable called `latex_file` that defaults to `None`;  if it has a filename, then output the latex code to that file, otherwise don't output latex

Reply:
it's not clear to me how to make your code work with different datasets; that needs to be documented in docstring